kratos/processes/structured_mesh_generator_process.cpp

```diff
@@ -232,6 +227,71 @@ namespace Kratos
 		}
 	}
 
+    void StructuredMeshGeneratorProcess::Check()
+    {
+        KRATOS_TRY
+
+        KRATOS_CHECK(CheckDomainGeometry());
+        KRATOS_CHECK(KratosComponents<Element>::Has(mElementName));
+
+        if ((mrGeometry.GetGeometryType() != GeometryData::KratosGeometryType::Kratos_Quadrilateral2D4) &
+            (mrGeometry.GetGeometryType() != GeometryData::KratosGeometryType::Kratos_Hexahedra3D8))
+            KRATOS_ERROR << "An unsupported geometry was given. Only Quadrilateral2D4 and Hexahedra3D8 are supported and given geometry is : " << mrGeometry << std::endl;
+
+        KRATOS_CHECK_NOT_EQUAL(mNumberOfDivisions, 0);
+
+        KRATOS_CATCH("")
+    }
+
+    bool StructuredMeshGeneratorProcess::CheckDomainGeometry() {
+        if (mrGeometry.GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Quadrilateral2D4) {
+            return CheckDomainGeometryConnectivityForQuadrilateral2D4();
+        }
+
+        else if (mrGeometry.GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Hexahedra3D8) {
+            return CheckDomainGeometryConnectivityForHexahedra3D8();
+        }
+
+        return true;
+    }
+
+    bool StructuredMeshGeneratorProcess::CheckDomainGeometryConnectivityForQuadrilateral2D4() {
+        return true; // TO DO: write check
+    }
+
+
+    bool StructuredMeshGeneratorProcess::CheckDomainGeometryConnectivityForHexahedra3D8() {
+        using tetrahedra_connectivity_in_cell_type = std::array<std::size_t, 4>;
```

Comment:
I understand that you are checking for degenerated hexahedron. Isn't it?

Reply:
I am constructing the tetrahedra the way you do it with the given geometry and checking for negative volumes. I just guessed (and checked in some cases) that this is an appropriate way to check this condition. We can think of a better way, nonetheless.